applications/RANSApplication/tests/cpp/test_vms_monolithic_k_based_condition.cpp

```diff
@@ -0,0 +1,345 @@
+#include "containers/model.h"
+#include "testing/testing.h"
+
+// Application includes
+#include "custom_utilities/test_utilities.h"
+#include "includes/cfd_variables.h"
+#include "rans_application_variables.h"
+
+namespace Kratos
+{
+namespace Testing
+{
+namespace
+{
+ModelPart& RansVMSMonolithicKBasedWall2D2N_SetUp(Model& rModel)
+{
+    const auto add_variables_function = [](ModelPart& rModelPart) {
+        rModelPart.AddNodalSolutionStepVariable(VELOCITY);
+        rModelPart.AddNodalSolutionStepVariable(PRESSURE);
+        rModelPart.AddNodalSolutionStepVariable(DENSITY);
+        rModelPart.AddNodalSolutionStepVariable(MESH_VELOCITY);
+        rModelPart.AddNodalSolutionStepVariable(KINEMATIC_VISCOSITY);
+        rModelPart.AddNodalSolutionStepVariable(ACCELERATION);
+        rModelPart.AddNodalSolutionStepVariable(EXTERNAL_PRESSURE);
+        rModelPart.AddNodalSolutionStepVariable(TURBULENT_KINETIC_ENERGY);
+    };
+
+    using namespace RansApplicationTestUtilities;
+
+    auto& r_model_part = CreateTestModelPart(
+        rModel, "Element2D3N", "RansVMSMonolithicKBasedWall2D2N", add_variables_function,
+        [](ModelPart::NodeType& rNode) {
+            rNode.AddDof(VELOCITY_X).SetEquationId(rNode.Id() * 4);
+            rNode.AddDof(VELOCITY_Y).SetEquationId(rNode.Id() * 4 + 1);
+            rNode.AddDof(VELOCITY_Z).SetEquationId(rNode.Id() * 4 + 2);
+            rNode.AddDof(PRESSURE).SetEquationId(rNode.Id() * 4 + 3);
+        },
+        1);
+
+    // set nodal historical variables
+    RandomFillNodalHistoricalVariable(r_model_part, VELOCITY, -10.0, 10.0);
+    RandomFillNodalHistoricalVariable(r_model_part, DENSITY, 2.0, 2.0);
+    RandomFillNodalHistoricalVariable(r_model_part, KINEMATIC_VISCOSITY, 1e-5, 1e-3);
+    RandomFillNodalHistoricalVariable(r_model_part, PRESSURE, 10.0, 100.0);
+    RandomFillNodalHistoricalVariable(r_model_part, MESH_VELOCITY, 1e-3, 1e-1);
+    RandomFillNodalHistoricalVariable(r_model_part, TURBULENT_KINETIC_ENERGY, 10.0, 40.0);
+    RandomFillNodalHistoricalVariable(r_model_part, ACCELERATION, 1.0, 1000.0);
+    RandomFillNodalHistoricalVariable(r_model_part, EXTERNAL_PRESSURE, 1.0, 1000.0);
+
+    RandomFillContainerVariable<ModelPart::ConditionsContainerType>(
+        r_model_part, NORMAL, -2.0, -1.0);
+
+    auto& r_condition = r_model_part.Conditions().front();
+    auto& r_element = r_model_part.Elements().front();
+    r_condition.SetValue(NEIGHBOUR_ELEMENTS, GlobalPointersVector<Element>{&r_element});
+
+    // set process info variables
+    auto& r_process_info = r_model_part.GetProcessInfo();
+    r_process_info.SetValue(WALL_VON_KARMAN, 3.1);
+    r_process_info.SetValue(WALL_SMOOTHNESS_BETA, 4.2);
+    r_process_info.SetValue(RANS_LINEAR_LOG_LAW_Y_PLUS_LIMIT, 12.0);
+    r_process_info.SetValue(TURBULENCE_RANS_C_MU, 0.09);
+
+    return r_model_part;
+}
+
+} // namespace
+
+KRATOS_TEST_CASE_IN_SUITE(RansVMSMonolithicKBasedWall2D2N_EquationIdVector, KratosRansFastSuite)
```

Comment:
Naming convention.

Reply:
This I would like to keep as it is because its a long name, having underscores gives little bit of readability :)